nixos/modules/security/wrappers/wrapper.c

```diff
@@ -177,6 +178,17 @@ int main(int argc, char **argv) {
         fprintf(stderr, "cannot readlink /proc/self/exe: %s", strerror(-self_path_size));
     }
 
+    unsigned int ruid, euid, suid, rgid, egid, sgid;
+    ASSERT(!getresuid(&ruid, &euid, &suid));
+    ASSERT(!getresgid(&rgid, &egid, &sgid));
+
+    // If true, then we did not benefit from setuid privilege escalation,
+    // where the original uid is still in ruid and different from euid == suid.
+    int no_suid = (ruid == euid) && (euid == suid);
```

Comment:
```suggestion
    int didnt_suid = (ruid == euid) && (euid == suid);
```

no_suid feels like this is a no SUID binary when in practice "it is", the difference is that we didn't actively set the UID because we are already that UID.

So the variable name should reflect this, `possess_already_target_uid`, or whatever. This makes the code more readable to the non-expert.